graphbolt/include/graphbolt/csc_sampling_graph.h

```diff
@@ -243,18 +245,14 @@ class CSCSamplingGraph : public torch::CustomClassHolder {
       PickFn pick_fn) const;
 
   /**
-   * @brief Build a CSCSamplingGraph from shared memory tensors.
-   *
-   * @param shared_memory_tensors A tuple of two share memory objects holding
-   * tensor meta information and data respectively, and a vector of optional
-   * tensors on shared memory.
+   * @brief Build a CSCSamplingGraph from a shared memory helper. This function
+   * takes ownership of the shared memory objects in the helper.
    *
+   * @param shared_memory_helper The shared memory helper.
    * @return A new CSCSamplingGraph on shared memory.
    */
-  static c10::intrusive_ptr<CSCSamplingGraph> BuildGraphFromSharedMemoryTensors(
-      std::tuple<
-          SharedMemoryPtr, SharedMemoryPtr,
-          std::vector<torch::optional<torch::Tensor>>>&& shared_memory_tensors);
+  static c10::intrusive_ptr<CSCSamplingGraph> BuildGraphFromSharedMemoryHelper(
```

Comment:
We may need to define it as a friend function or add an interface to set the shared memory pointers. I don't want to add the interface because we only set the shared memory pointers when loading the graph from shared memory.

Reply:
Yeah, I agree. But I cannot think of a better solution.